src/main/java/mocacong/server/itegration/auth/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -0,0 +1,33 @@
+package mocacong.server.itegration.auth;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Component
+public class AuthenticationPrincipalArgumentResolver implements HandlerMethodArgumentResolver {
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public AuthenticationPrincipalArgumentResolver(final JwtTokenProvider jwtTokenProvider) {
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    public boolean supportsParameter(MethodParameter parameter) {
+        return false;
+    }
```

Comment:
`LoginEmail`은 불필요한 dto라서 없애는 것에 동의하되, `LoginUserEmail`까지 없애면 안된다고 생각해요. `supportsParameter` 메서드에서 바인딩할 객체를 지정해주어야 컨트롤러에서 해당 어노테이션 (ex. LoginUserEmail)이 붙은 api에서 인증여부를 확인해줄 수 있기 때문이에요. 

 ```suggestion
    @Override
    public boolean supportsParameter(MethodParameter parameter) {
         return parameter.hasParameterAnnotation(LoginUserEmail.class);
    }
```
위와 같은 방법은 어떤가요? 참고로 클래스명은 어떻든 상관없습니다.

Reply:
```java
@Override
    public boolean supportsParameter(MethodParameter parameter) {
        return parameter.getParameterType().equals(LoginUserEmail.class);
    }
```
혹시 이런 방식도 괜찮나요? 케이님이 주신 코드를 적용하려면 기존에 있었던 `LoginUserEmail` 을 수정해야 할 거 같은데 제가 제시한 코드는 파라미터의 타입이 `LoginUserEmail` 인지 여부를 확인하는 거라 결국 케이님이 제시하신 거랑 같은 동작을 한다고 생각해서 질문드립니다!

Reply:
네 동작만 제대로 되면 괜찮을 듯합니다! 큰 문제 없어보일 거 같아요. 지금은 아직 인증 필요한 api도 없어서 정확하게 체크하긴 어려울 듯한데 혹시나 만약 문제생기면 나중에 고치죠~

Reply:
넵 알겠습니다~